pkgs/top-level/all-packages.nix

```diff
@@ -6365,7 +6365,7 @@ with pkgs;
   cudaPackages_11_6 = callPackage ./cuda-packages.nix { cudaVersion = "11.6"; };
   cudaPackages_11_7 = callPackage ./cuda-packages.nix { cudaVersion = "11.7"; };
   cudaPackages_11_8 = callPackage ./cuda-packages.nix { cudaVersion = "11.8"; };
-  cudaPackages_11 = cudaPackages_11_7;
+  cudaPackages_11 = cudaPackages_11_8;
```

Comment:
Why does this belong in this PR? Can't we do it separately?

Reply:
We can! Originally I had included it for this reason (I updated the OP)

> On changing the default `cudatoolkit` from 11.7 to 11.8:
>
> - CUDA 11 only supports Ada and Hopper as of 11.8.
> - If we do not bump to 11.8, we have two options:
>   - Either we must remove Ada and Hopper because they are not supported by the default cudaotoolkit and trying to get NVCC to target those architectures will throw errors
>   - Or we dynamically figure out which architectures are supported by different versions of the cudatoolkit. I did something similar in a PR here for CUDNN: https://github.com/NixOS/nixpkgs/pull/217363.
> - The latter is, in my opinion, a better long-term solution and ensures that the end-user can build for whichever architectures are supported by their chosen version of CUDA and not some list hard-coded in nixpkgs.

In the rewrite I'll push soon, it shouldn't be necessary.